ql/src/java/org/apache/hadoop/hive/ql/parse/DDLSemanticAnalyzer.java

```diff
@@ -2571,11 +2567,10 @@ private void analyzeDescDatabase(ASTNode ast) throws SemanticException {
       throw new SemanticException("Unexpected Tokens at DESCRIBE DATABASE");
     }
 
-    DescDatabaseDesc descDbDesc = new DescDatabaseDesc(ctx.getResFile(),
-        dbName, isExtended);
+    DescDatabaseDesc descDbDesc = new DescDatabaseDesc(ctx.getResFile(), dbName, isExtended);
     inputs.add(new ReadEntity(getDatabase(dbName)));
-    rootTasks.add(TaskFactory.get(new DDLWork(getInputs(), getOutputs(), descDbDesc)));
-    setFetchTask(createFetchTask(descDbDesc.getSchema()));
+    rootTasks.add(TaskFactory.get(new DDLWork2(getInputs(), getOutputs(), descDbDesc)));
+    setFetchTask(createFetchTask(DESC_DATABASE_SCHEMA));
```

Comment:
I think the `schema` should works similarily to earlier; how about asking the DDLWork for the schema? it could look up based on the desc

Reply:
will do